e2e_tests/aws_ec2/utils.py

```diff
@@ -0,0 +1,104 @@
+import boto3
+import logging
+import requests
+from time import sleep
+import subprocess
+
+logger = logging.getLogger('bentoml.test')
+
+
+def get_instance_ip_from_id(instance_id, api, region):
+    ec2_client = boto3.client("ec2", region)
+    response = ec2_client.describe_instances(InstanceIds=[instance_id])
+    ip_address = response["Reservations"][0]["Instances"][0]["PublicIpAddress"]
+    url = "http://" + ip_address + ":" + "5000" + api
+    return url
+
+
+def get_addresses_from_target_group(target_group_arn, api, region):
+    healthy_target_status = "healthy"
+    eb_client = boto3.client("elbv2", region)
+
+    all_targets = eb_client.describe_target_health(TargetGroupArn=target_group_arn)
+    all_addresses = []
+    for instance in all_targets["TargetHealthDescriptions"]:
+        health = instance["TargetHealth"]["State"]
+        if not health == healthy_target_status:
+            # if any target is unhealthy,wait.
+            return None
+        id_ = instance["Target"]["Id"]
+        all_addresses.append(get_instance_ip_from_id(id_, api, region))
+    return all_addresses
+
+
+def wait_for_healthy_targets(name, namespace, region):
```

Comment:
add a bit inline comments here to explain what this function does?

Reply:
Added